src/v/archival/ntp_archiver_service.cc

```diff
@@ -2951,46 +2951,129 @@ size_t ntp_archiver::get_local_segment_size() const {
     return log_segment_size;
 }
 
-ss::future<bool>
-ntp_archiver::prepare_transfer_leadership(ss::lowres_clock::duration timeout) {
-    _paused = true;
-
+ss::future<bool> ntp_archiver::pause(ss::lowres_clock::duration timeout) {
     ss::gate::holder holder(_gate);
+
+    vlog(
+      _rtclog.info,
+      "Pausing archiver; active={}",
+      _uploads_active.current() != 0);
+
+    const auto deadline = ss::lowres_clock::now() + timeout;
+    bool success = true;
+
     try {
-        co_await ss::get_units(_uploads_active, 1, timeout);
+        // Phase 1: disable the housekeeping jobs and if the currently running
+        // job is managed by the archiver, then wait for it to complete
+        if (auto workflow_wrap = _parent.cloud_housekeeping_workflow();
+            workflow_wrap) {
+            _scrubber->set_enabled(false);
+            _local_segment_merger->set_enabled(false);
```

Comment:
Should these be checked if they are valid before enabling them? Same when unpausing.

Reply:
Yes, they should. Very good point.

Reply:
Fixed.